internal/imodels/imodels.go

```diff
@@ -0,0 +1,164 @@
+package imodels
+
+import (
+	"log"
+
+	"github.com/google/osv-scalibr/extractor"
+	"github.com/google/osv-scalibr/extractor/filesystem/os/apk"
+	"github.com/google/osv-scalibr/extractor/filesystem/os/dpkg"
+	"github.com/google/osv-scalibr/extractor/filesystem/os/rpm"
+	"github.com/google/osv-scalibr/extractor/filesystem/sbom/cdx"
+	"github.com/google/osv-scalibr/extractor/filesystem/sbom/spdx"
+	"github.com/google/osv-scanner/internal/imodels/ecosystem"
+	"github.com/google/osv-scanner/internal/lockfilescalibr/vcs/gitrepo"
+	"github.com/google/osv-scanner/pkg/models"
+
+	scalibrosv "github.com/google/osv-scalibr/extractor/filesystem/osv"
+)
+
+var sbomExtractors = map[string]struct{}{
+	spdx.Extractor{}.Name(): {},
+	cdx.Extractor{}.Name():  {},
+}
+
+var gitExtractors = map[string]struct{}{
+	gitrepo.Extractor{}.Name(): {},
+}
+
+var osExtractors = map[string]struct{}{
+	dpkg.Extractor{}.Name(): {},
+	apk.Extractor{}.Name():  {},
+	rpm.Extractor{}.Name():  {},
+}
+
+// PackageInfo represents a package found during a scan. This is generally
+// converted directly from the extractor.Inventory type, with some restructuring
+// for easier use within osv-scanner itself.
+type PackageInfo struct {
+	Name      string // Name will be SourceName matching the osv-schema
+	Version   string
+	Ecosystem ecosystem.Parsed
+
+	Location   string // Contains Inventory.Locations[0]
+	SourceType SourceType
+
+	Commit     string
+	Repository string
+
+	// For package sources
+	DepGroups []string
+
+	// For OS packages
```

Comment:
can you add a bit more details to the comments here to explain that for OS packages where the name is the source package, this is typically the binary name?